<?xml version="1.0" ?>
<launch>

    <!-- Arguments -->
    <arg name="semantic" default="true" doc="Use Voxblox++ for semantic scene reconstruction" />
    <arg name="launch_rviz" default="false"/>
    <arg name="simulation_mode" default="false"/>

    <!-- ======== Launch everything ======== -->

    <!-- Load parameters -->
    <rosparam command="load" file="$(find grasp_demo)/config/grasp_demo.yaml" ns="manipulator/moma_demo"/>

    <!-- Launch camera -->
    <include file="$(find realsense2_camera)/launch/rs_camera.launch" unless="$(arg simulation_mode)">
        <arg name="depth_width" value="640"/>
        <arg name="depth_height" value="480"/>
        <arg name="depth_fps" value="30"/>
        <arg name="enable_pointcloud" value="true"/>
        <arg name="filters" value="temporal"/>
        <arg name="tf_prefix" value="wrist_camera"/>
        <arg name="publish_tf" value="true"/>
    </include>

    <!-- Bringup robot -->
    <include unless="$(arg simulation_mode)" file="$(find moma_bringup)/launch/panda_moveit_real.launch" ns="manipulator">
        <arg name="launch_rviz" value="false"/>
    </include>
    <include if="$(arg simulation_mode)" file="$(find moma_gazebo)/launch/panda_example.launch">
        <arg name="rviz" value="false"/>
        <arg name="world_name" value="$(find moma_gazebo)/worlds/table_top_1.world"/>
        <arg name="panda_z_position" value="0.8"/>
        <arg name="use_fixed_realsense" value="true"/>
    </include>

    <!-- Launch action scripts -->
    <node pkg="grasp_demo" type="reset.py" name="reset" ns="manipulator" output="screen"/>
    <node pkg="grasp_demo" type="reconstruct_scene.py" name="scan_action_node" args="$(arg semantic)" ns="manipulator" output="screen" />
    <node pkg="grasp_demo" type="plan_grasp.py" name="plan_grasp" ns="manipulator" output="screen"/>
    <node pkg="grasp_demo" type="execute_grasp.py" name="execute_grasp" ns="manipulator" output="screen"/>
    <node pkg="grasp_demo" type="drop_object.py" name="drop_object" ns="manipulator" output="screen"/>

    <node pkg="grasp_demo" type="run_plan.py" name="grasp_demo" ns="manipulator" output="screen"/>

    <!-- Open rviz if desired -->
    <node if="$(arg launch_rviz)" type="rviz" name="rviz" pkg="rviz" args="-d $(find grasp_demo)/config/grasp_demo.rviz" ns="manipulator"/>

</launch>
